api/src/main/java/edu/cornell/mannlib/vitro/webapp/dynapi/components/AbstractOperation.java

```diff
@@ -0,0 +1,48 @@
+package edu.cornell.mannlib.vitro.webapp.dynapi.components;
+
+
+import edu.cornell.mannlib.vitro.webapp.utils.configuration.Property;
+import org.apache.commons.logging.LogFactory;
+import edu.cornell.mannlib.vitro.webapp.dynapi.OperationData;
+import org.apache.commons.logging.Log;
+
+public abstract class AbstractOperation implements Operation{
+
+
+    protected Parameters requiredParams = new Parameters();
+    protected Parameters providedParams = new Parameters();
+
+    @Property(uri = "https://vivoweb.org/ontology/vitro-dynamic-api#requiresParameter")
+    public void addRequiredParameter(Parameter param) {
+        requiredParams.add(param);
+    }
+
+    @Property(uri = "https://vivoweb.org/ontology/vitro-dynamic-api#providesParameter")
+    public void addProvidedParameter(Parameter param) {
```

Comment:
Not all operations provide new parameters.
I added this table a few days ago https://docs.google.com/document/d/1n7gSf_cSDS5mbTI4HwVGj-2sQd5if0177Nrdtri9BtM/edit#heading=h.o146rfjwwctn
 There were some fixes to the ontology that aren't merged yet https://github.com/vivo-project/Vitro/pull/265